### DISCUSSION TITLE
Permissions issues after upgrade to Big Sur 11.4

### DISCUSSION THREAD

**Discussion by Paul-Morris** [Homebrew/discussions#1567] [Everyday usage]
I'm getting a lot of permissions errors when trying to brew upgrade. Is this something I can fix? I should have waited a while before installing the update for Big Sur 11.4

`==> Installing dependencies for curl: libidn2, openssl@1.1, nghttp2, openldap and zstd
==> Installing curl dependency: libidn2
==> Pouring libidn2--2.3.1.big_sur.bottle.tar.gz
cp: /private/tmp/d20210527-48741-1v9ynwe/libidn2/.: unable to copy extended attributes to /usr/local/Cellar/libidn2/.: Bad file descriptor
cp: /usr/local/Cellar/libidn2/./2.3.1: Permission denied`

**Comment by carlocab** [comment]
What's the output of `brew config` and `brew doctor`?

**Reply by Paul-Morris** [reply]
**brew config**
HOMEBREW_VERSION: 3.1.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 76cfe88b7628a8010bf98115627f6aced4ff8870
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 737dc591136eca909d2a442072ebfe02b0307f10
Core tap last commit: 70 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 12.0.5 build 1205
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.4-x86_64
CLT: 12.5.0.22.9
Xcode: N/A
Rosetta 2: true


**brew doctor**
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  php@7.2

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  pkg-config
  php
  vde
  libevent
  unbound
  libusb

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

**Comment by Paul-Morris** [comment]
Note: I ended up fixing this by using:

**sudo chown -R $(whoami) /usr/local/***

I hope I didn't break anything but everything is working fine so far
